src/System.IO.Abstractions.Extensions/IDirectoryInfoExtensions.cs

```diff
@@ -3,25 +3,103 @@
     public static class IDirectoryInfoExtensions
     {
         /// <summary>
-        /// Get an <see cref="IDirectoryInfo"/> for the specified sub-directory <paramref name="name"/>
```

Comment:
Also can you please move the new methods at the bottom of the file instead of the top so the merge looks cleaner